Sources/MCPToolMacros/Plugin.swift

```diff
@@ -0,0 +1,10 @@
+import SwiftCompilerPlugin
+import SwiftSyntaxMacros
+
+@main
+struct MCPToolMacrosPlugin: CompilerPlugin {
```

Comment:
Naming: The compiler plugin type uses the same name as the library product (`MCPToolMacrosPlugin`). Consider renaming the type to `MCPToolMacrosCompilerPlugin` to avoid confusion between product and type.